source/patterns/@aws-solutions-constructs/core/lib/glue-job-helper.ts

```diff
@@ -99,41 +107,55 @@ export function buildGlueJob(scope: Construct, props: BuildGlueJobProps): [glue.
         'it is recommended to use "WorkerType" or  "NumberOfWorkers"');
       }
 
-      return deployGlueJob(scope, props.glueJobProps, props.database!, props.table!, props.outputDataStore!, props.etlCodeAsset);
+      const deployGlueJobResponse =
+        deployGlueJob(scope, props.glueJobProps, props.database!, props.table!, props.outputDataStore!, props.etlCodeAsset);
+      return {
+        job: deployGlueJobResponse.job,
+        role: deployGlueJobResponse.role,
+        bucket: deployGlueJobResponse.bucket,
+        loggingBucket: deployGlueJobResponse.loggingBucket };
     } else {
       throw Error('Either glueJobProps or existingCfnJob is required');
     }
   } else {
-    return [props.existingCfnJob, Role.fromRoleArn(scope, 'ExistingRole', props.existingCfnJob.role)];
+    return { job: props.existingCfnJob, role: Role.fromRoleArn(scope, 'ExistingRole', props.existingCfnJob.role)};
   }
 }
 
+export interface DeployGlueJobResponse {
+  readonly job: glue.CfnJob,
+  readonly role: IRole,
+  readonly bucket?: Bucket,
+  readonly loggingBucket?: Bucket,
+}
+
 export function deployGlueJob(scope: Construct, glueJobProps: glue.CfnJobProps, database: glue.CfnDatabase, table: glue.CfnTable,
-  outputDataStore: SinkDataStoreProps, etlCodeAsset?: s3assets.Asset): [glue.CfnJob, IRole, [Bucket, (Bucket | undefined)?]] {
+  outputDataStore: SinkDataStoreProps, etlCodeAsset?: s3assets.Asset): DeployGlueJobResponse {
+  // TODO - replicate [glue.CfnJob, IRole, [Bucket, (Bucket | undefined)?]]
```

Comment:
I'm not sure what this TODO means? Let's add some more details so it makes sense to the next person, or clean it up!

Reply:
It just got left behind, the work was done. I will remove.